pkg/api/message/v1/context/context.go

```diff
@@ -61,3 +67,22 @@ func parseVersionHeaderValue(vals []string) (name string, version string, full s
 	}
 	return
 }
+
+func (ri *requesterInfo) isSupportedClient() bool {
+	// Only version-gate on production
+	if os.Getenv("ENV") != "production" {
```

Comment:
The perf impact shouldn't be particularly noticeable ([75ns/op](https://medium.com/@morfien101/are-environment-variables-in-go-fast-3d9dde220b6f)), but given this value doesn't change and `os.GetEnv` isn't free we should probably move this out of the function so we only have to do the lookup once on startup.

Reply:
I did actually check the same thing, but I found it's actually a decent amount of plumbing to get the config into everything that calls this, and I decided to keep this PR small given the sensitive timing

Reply:
Fair enough. I'm not losing sleep over 75ns